### PR TITLE
feat: add Text component to `react-text`

### PR DESCRIPTION
**Pull request by layershifter**
This PR adds an initial version of `Text` in `@fluentui/react-text`.

### CODE REVIEW

File: packages/react-theme-provider/src/useTheme.ts [hunk 1]
```diff
@@ -17,6 +17,6 @@ function useCompatTheme(): ITheme | undefined {
 export const useTheme = (): Theme => {
   const theme = useContext(ThemeContext);
   const legacyTheme = useCompatTheme();
-
+  console.log(theme);
```
Comment (layershifter) [inline packages/react-theme-provider/src/useTheme.ts:20]:
```suggestion

```

File: packages/react-text/src/components/Text/useText.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { makeMergeProps, getSlots, resolveShorthandProps } from '@fluentui/react-compose/lib/next/index';
+import { TextProps } from './Text.types';
+
+const mergeProps = makeMergeProps();
+
+/**
+ * Define the render function. Given the state of a text, renders it.
+ */
+export const renderText = (state: TextProps) => {
+  const { slots, slotProps } = getSlots(state, [] /* there is no slots in Text */);
```
Comment (khmakoto) [inline packages/react-text/src/components/Text/useText.tsx:11]:
```suggestion
  const { slots, slotProps } = getSlots(state, [] /* there are no slots in Text */);
```

File: packages/react-text/src/components/Text/useText.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { makeMergeProps, getSlots, resolveShorthandProps } from '@fluentui/react-compose/lib/next/index';
+import { TextProps } from './Text.types';
+
+const mergeProps = makeMergeProps();
+
+/**
+ * Define the render function. Given the state of a text, renders it.
+ */
+export const renderText = (state: TextProps) => {
```
Comment (khmakoto) [inline packages/react-text/src/components/Text/useText.tsx:10]: Should we move `renderText` into a separate file?

File: packages/react-text/src/components/Text/Text.types.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { BaseSlots, ComponentProps, SlotProps } from '@fluentui/react-compose';
+import { RecursivePartial } from '@fluentui/theme';
+import * as React from 'react';
+
+export interface TextProps extends ComponentProps, React.ImgHTMLAttributes<HTMLSpanElement> {
```
[the remark this answers is not in the document]
Comment (layershifter) [reply packages/react-text/src/components/Text/Text.types.ts:5]: Oops, copy-paste issue, good catch 👍

File: packages/react-text/src/components/Text/useText.tsx (hunk 3 above)
Comment (layershifter) [reply packages/react-text/src/components/Text/useText.tsx:10]: Moved 👍

**Review by khmakoto — APPROVED** [review]

**Comment by khmakoto** [comment]
Seems like you need to run `yarn update-api` for the build to pass.

File: packages/react-text/src/components/Text/useTextClasses.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { makeClasses } from '@fluentui/react-theme-provider';
+
+const GlobalClassNames = {
+  root: 'ms-Text',
+};
+
+export const useTextClasses = makeClasses({
```
Comment (dzearing) [inline packages/react-text/src/components/Text/useTextClasses.ts:8]: this looks great, and also, I just merged `makeVariantClasses` should we want these to be easily definable and extendable in the theme. Can sync later on this.

**Review by dzearing — APPROVED** [review]
looks great!

**Pull request merged**